keras/models.py

```diff
@@ -97,13 +99,19 @@ def get_json_type(obj):
 
     from . import __version__ as keras_version
 
-    # If file exists and should not be overwritten.
-    if not overwrite and os.path.isfile(filepath):
-        proceed = ask_to_proceed_with_overwrite(filepath)
-        if not proceed:
-            return
+    opened_new_file = not isinstance(filepath, h5py.File)
```

Comment:
I suggest only setting this flag to `True` after actually creating the file (since it is used to call `f.close()` at the end).

You can start with `if not isinstance(...)`